src/ndarray/ndarray.cc

```diff
@@ -614,8 +614,7 @@ NDArray &NDArray::operator/=(const real_t &src) {
 }
 
 void NDArray::Save(dmlc::Stream *strm) const {
-  // save shape
-  shape_.Save(strm);
```

Comment:
There is a reserved field on the dump file format that can be used for version checking